### PR TITLE
Make commodity code is optional in the measure_condition_dialog.

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link
https://transformuk.atlassian.net/browse/HOTT-3643

### What?
Make commodity code optional in the measure_condition_dialog.

NOTE: to make it clear that a particular dialog condition will be applied to any goods_nomenclature_item_id,
we are using the symbol '*' instead of empty space:

`goods_nomenclature_item_id: '*'`
(it reminds the * used in the Regular expression)

### Why?
It would be a very unwieldy, large and unmanageable configuration file, if we had to supply every permutation of commodity plus measure type plus additional_code (literally thousands of rows).

- [x ] Reviewed view changes with stake holders
- [x] Validated mobile responsive behaviour of view changes

### CODE REVIEW

**Comment by alexdesi** [comment]
There is a broken test, I believe it is a flaky one. I am working on it now.

File: app/models/measure_condition_dialog.rb [hunk 1]
```diff
@@ -48,15 +47,30 @@ def build(declarable, measure, anchor = 'import')
...
 
     def config_for(goods_nomenclature_item_id, additional_code, measure_type_id)
-      measure_condition_to_augment = config.find do |entry|
+      measure_condition_to_augment = conditions_for_specific_nomenclature_items.find do |entry|
         entry['goods_nomenclature_item_id'] == goods_nomenclature_item_id &&
           entry['additional_code'] == additional_code &&
           entry['measure_type_id'] == measure_type_id
       end
 
+      # If there are not matches, it applies the `*` search:
```
Comment (alexdesi) [inline app/models/measure_condition_dialog.rb]: fix the comment

File: db/measure_condition_dialog_config.yaml [hunk 2]
```diff
@@ -1,30 +1,10 @@
 ---
-- goods_nomenclature_item_id: '2203000100'
+- goods_nomenclature_item_id: '*'
```
Comment (alexdesi) [inline db/measure_condition_dialog_config.yaml]: remove this line

File: spec/models/measure_condition_dialog_spec.rb [hunk 3]
```diff
@@ -56,6 +56,28 @@
       end
     end
 
+    context 'when the config file specifies generic goods nomenclatures (= no specific goods_nomenclature_item_id)' do
+      # before do
```
Comment (willfish) [inline spec/models/measure_condition_dialog_spec.rb]: Don't think you want this stubbed?

File: app/models/measure_condition_dialog.rb [hunk 4]
```diff
@@ -48,15 +47,30 @@ def build(declarable, measure, anchor = 'import')
...
+      measure_condition_to_augment ||= conditions_for_unspecified_nomenclature_items.find do |entry|
+        entry['additional_code'] == additional_code &&
+          entry['measure_type_id'] == measure_type_id
+      end
+
       measure_condition_to_augment || DEFAULT_CONFIG_ENTRY
     end
 
+    def conditions_for_specific_nomenclature_items
+      config.select { |entry| entry['goods_nomenclature_item_id'].present? }
```
Comment (willfish) [inline app/models/measure_condition_dialog.rb:67]:
```suggestion
      config.select { |entry| entry['goods_nomenclature_item_id'].present? && entry['goods_nomenclature_item_id'] != UNFILTERED_GOODS_NOMENCLATURE_SYNTAX }
```

**Review by willfish — DISMISSED** [review]

**Review by willfish — APPROVED** [review]

**Review by willfish — DISMISSED** [review]

**Review by willfish — APPROVED** [review]

**Pull request merged**